### PR TITLE
Bug 1999261: filter out KubeAPIAuth when logging CNI requests (improved)

### PR DESCRIPTION
**Pull request by ricky-rav**
minimize marshalling operations when filtering out KubeAPIAuth in logged CNI requests

Original PR (merged): https://github.com/ovn-org/ovn-kubernetes/pull/2479

This patches addresses the changes requested in: https://github.com/openshift/ovn-kubernetes/pull/731  (... which will be closed)

### CODE REVIEW

File: go-controller/pkg/cni/cni.go [hunk 1]
```diff
@@ -136,27 +135,22 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, podLister corev1listers.PodL
...
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal pod request response: %v", err)
-	}
-
-	return responseBytes, nil
+	return response, nil
 }
 
-func (pr *PodRequest) cmdDel() ([]byte, error) {
+func (pr *PodRequest) cmdDel() (*Response, error) {
```
Comment (danwinship) [inline go-controller/pkg/cni/cni.go:141]: `cmdDel` and `cmdCheck` never return any actual response data. You can just remove the first argument from them

File: go-controller/pkg/cni/types.go [hunk 2]
```diff
@@ -79,6 +81,32 @@ type Response struct {
...
+	var err error
+
+	if response == nil {
+		return nil, nil
+	}
+
+	noAuth := struct {
+		Result    *current.Result
+		PodIFInfo *PodInterfaceInfo
+	}{Result: response.Result, PodIFInfo: response.PodIFInfo}
```
Comment (danwinship) [inline go-controller/pkg/cni/types.go:100]: if you look at the details of `cmdAdd`, every `Response` has only one of `Result` and `PodIFInfo` set. So you could just marshal and log whichever of them is non-`nil`

File: go-controller/pkg/cni/cni.go (hunk 1 above)
Comment (ricky-rav) [reply go-controller/pkg/cni/cni.go]: done!

File: go-controller/pkg/cni/types.go (hunk 2 above)
Comment (ricky-rav) [reply go-controller/pkg/cni/types.go]: That's also true! I just pushed a commit with the changes you requested.

**Pull request merged**